test/e2e_node/critical_pod_test.go

```diff
@@ -111,6 +124,24 @@ var _ = SIGDescribe("CriticalPod [Serial] [Disruptive] [NodeFeature:CriticalPod]
 	})
 })
 
+func createSyncInNamespaceFromPod(f *framework.Framework, pod *v1.Pod) *v1.Pod {
+	p := f.PodClientNS(pod.Namespace).Create(pod)
+	framework.ExpectNoError(e2epod.WaitTimeoutForPodReadyInNamespace(f.ClientSet, p.Name, p.Namespace, framework.PodStartTimeout))
+	// Get the newest pod after it becomes running and ready, some status may change after pod created, such as pod ip.
+	p, err := f.PodClientNS(pod.Namespace).Get(context.TODO(), p.Name, metav1.GetOptions{})
+	framework.ExpectNoError(err)
+	return p
+}
+
+func deleteSyncInNamespace(f *framework.Framework, name, namespace string, options metav1.DeleteOptions, timeout time.Duration) {
+	err := f.PodClientNS(namespace).Delete(context.TODO(), name, options)
+	if err != nil && !apierrors.IsNotFound(err) {
+		framework.Failf("Failed to delete pod %q in namespace %q: %v", name, namespace, err)
+	}
+	gomega.Expect(e2epod.WaitForPodToDisappear(f.ClientSet, namespace, name, labels.Everything(),
```

Comment:
`WaitForPodToDisappear` already does a poll loop with the framework standard timeouts, we shouldn't wrap it in another loop
```
func WaitForPodToDisappear(c clientset.Interface, ns, podName string, label labels.Selector, interval, timeout time.Duration) error {
	return wait.PollImmediate(interval, timeout, func() (bool, error) {
```

Reply:
good point, but in this specific case (unless I'm missing something!) we don't wrap it directly or indirectly, so keeping the code as is for now.

Reply:
lol, sorry, I've mentally tied gomega.Expect with the 2.time.Second, timeout, 
apologies

Reply:
no problem at all! it's still a good point in general and I learned something!